src/libcollections/dlist.rs

```diff
@@ -367,12 +297,8 @@ impl<T> DList<T> {
         it.insert_next(elt);
     }
 
-    /// Merges `other` into this `DList`, using the function `f`.
-    ///
-    /// Iterates both `DList`s with `a` from self and `b` from `other`, and
-    /// put `a` in the result if `f(a, b)` is true, and otherwise `b`.
-    ///
-    /// This operation should compute in O(max(N, M)) time.
+    /// Deprecated: Use custom methods on IterMut.
+    #[deprecated = "Use custom methods on IterMut"]
     pub fn merge<F>(&mut self, mut other: DList<T>, mut f: F) where F: FnMut(&T, &T) -> bool {
```

Comment:
Also if you're interested in really complex DList manipulation, consider checking out my [experimental DList with a full Cursor API](http://gankro.github.io/collect-rs/collect/proto/dlist/index.html), which I think is the more correct solution to this kind of complex manipulation problem.